mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -2107,14 +2108,46 @@ static FailureOr<rock::TransformMapAttr> removeUpperDimsFromMap(
           origLowerBounds[lowerDim] = origUpperBounds[upperDim];
         }
         for (auto [dim, subDimInfo] : removedSubDims) {
+          LLVM_DEBUG(llvm::dbgs() << "copying removedSubDimInfo from:" << dim
+                                  << " to:" << upperToLower[dim] << "\n");
           newRemovedSubDims[upperToLower[dim]] = subDimInfo;
         }
-        [[fallthrough]];
+        llvm::copy(tr.getParams(), std::back_inserter(args.params));
+        break;
       }
-      default: {
+      case TransformType::Pad: {
+        for (auto [idx, upperDim] : llvm::enumerate(tr.getUpperDims())) {
```

Comment:
... why do we need this case?

Also, I'm dubious about leaving this rather specific assumption in a generically-named function without any warnings that you're about to cause problems.

Reply:
a sub-tile that d tile a padded region will need to index into the padded region as it was unpadded -- i.e. fully materialized region with padded value.

I dont see any problem with that given how our padding works. Im happy to entertain an example if you have one that is problematic.

Since this is blocking any reduction blockwise fusion, I can put a option to enable handling the padding to the function and use it explicitly

Reply:
> ... why do we need this case? 

When we tile a partial reduction axis, we need to pad it such that it has a exact split in block & thread tile. When that happens, we d be tiling the padded region. As you know, block & thread tile it is irrelevant that grid coordinates had padding in them -- as we would have tile sizes as if they were unpadded.